### PR TITLE
test/e2e: metric proxy setup is blocking tests on HA control plane

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
The metrics proxy setup code is  assuming there is exactly two pods that start with `kube-controller-mangaer-` and `kube-scheduler-`. This assumption is true when there is only 1 control plane and one instance of each of these components. However, if the cluster is running an HA control plane with 3 instances of each, this check will block forever. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by andrewsykim — COMMENTED** [review]
/assign @aojea @knight42

File: test/e2e/framework/metrics/metrics_proxy.go [hunk 1]
```diff
@@ -68,8 +68,8 @@ func SetupMetricsProxy(c clientset.Interface) error {
 				})
 			}
 		}
-		if len(foundComponents) != 2 {
-			klog.Infof("Only %d components found. Will retry.", len(foundComponents))
+		if len(foundComponents) < 2 {
```
Comment (andrewsykim) [inline test/e2e/framework/metrics/metrics_proxy.go:71]: In a 3 node control plane, this value is 6, so we poll for 5 minutes unnecessarily.

**Comment by aojea** [comment]
/hold
the idea is to remove the metrics proxy https://github.com/kubernetes/kubernetes/pull/102050

**Comment by andrewsykim** [comment]
Fine by me, but do we expect that to landing v1.22? Otherwise this will impact test suites against HA control plane

**Comment by aojea** [comment]
> Fine by me, but do we expect that to landing v1.22? Otherwise this will impact test suites against HA control plane

if we don't decide today feel free to remove the hold, but this is causing 5 minutes delay for openshift clusters too

**Comment by andrewsykim** [comment]
ack, thanks @aojea !

**Comment by BenTheElder** [comment]
I think just one remaining topic there, which may be a simple "no it's fine" or may be a "ok yes we need to make sure we fix that next / now" ... https://github.com/kubernetes/kubernetes/pull/102050#pullrequestreview-697672695

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/pull/102050 merged. the metrics proxy is removed.

**Review by andrewsykim — COMMENTED** [review]
Thanks all! 

/close

**Pull request closed without merging**